crates/compiler/gen_llvm/src/llvm/build.rs

```diff
@@ -919,6 +919,51 @@ impl<'a, 'ctx, 'env> Env<'a, 'ctx, 'env> {
         call.set_call_convention(C_CALL_CONV);
     }
 
+    pub fn call_dbg(
+        &self,
+        env: &Env<'a, 'ctx, 'env>,
+        location: BasicValueEnum<'ctx>,
+        message: BasicValueEnum<'ctx>,
+    ) {
+        let function = self.module.get_function("roc_dbg").unwrap();
+
+        let loc = match env.target_info.ptr_width() {
+            PtrWidth::Bytes4 => {
+                // we need to pass the location by reference, but we currently hold the value.
+                let alloca = env
+                    .builder
+                    .new_build_alloca(location.get_type(), "alloca_dbg_location");
+                env.builder.new_build_store(alloca, location);
+                alloca.into()
+            }
+            PtrWidth::Bytes8 => {
+                // string is already held by reference
+                location
+            }
```

Comment:
Feels like we should factor this out given the duplication here, and I know this is duplicated elsewhere in the file.

Reply:
done